### PR TITLE
[Intel]: add loop pipelining for prefetch in the pass pipeline

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

**Comment by chengjunlu** [comment]
Basically looks good to me.
The prefetching lowering distributed the prefetching tiles to the multiple threads as same as the load 2D lowering.
It may have some wasting in the prefetching instruction because multiple prefetching ops may prefetch the same address into the cache line.

There is an alternative tiling for prefetching since we don't care about the return value from it.
We used the terminology of "co-operative" prefetching. The tiling heuristic are implemented here: https://github.com/intel/intel-xpu-backend-for-triton/blob/3541627120d23258baac9d633754b02940d3cf57/lib/Conversion/TritonGPUToLLVM/LoadStoreOpToLLVM.cpp#L152

The good part of it:

- The "co-operative" prefetching saves the number of the prefetching instructions (Each prefetching instruction to use the maximum HW's capability.). 

The dis-advantage part of it:

- It is only friendly to the L3 cache. The L1 cache (close to each XeCore) is not tiling in the same way.

File: third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp [hunk 1]
```diff
@@ -164,6 +165,125 @@ struct LoadStoreConversionBase {
...
+        LoadStoreConversionBase(targetInfo, axisAnalysisPass) {}
+
+  LogicalResult
+  matchAndRewrite(triton::gpu::intel::PrefetchOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    Value ptr = op.getPtr();
+    if (isTensorPointerType(ptr.getType()))
+      return rewriteTensorPointerPrefetch(op, adaptor, rewriter);
+
+    assert(false && "Unexpected prefetch operation on 'regular' ptr");
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:189]: nit: unreachable

**Review by whitneywhtsang — APPROVED** [review]
Please add a lit test.

**Comment by etiotto** [comment]
> Please add a lit test.

Added

File: third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp (hunk 1 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/LoadStoreOpToLLVM.cpp:189]: done

**Pull request merged**